File: CollabTodoDesktop/Models/Task.cs
using System;

namespace CollabTodoDesktop.Models;

/// <summary>
/// 작업 도메인 모델
/// Python 버전의 Task dataclass와 동일한 구조
/// </summary>
public record Task
{
    public int Id { get; init; }
    public int ProjectId { get; init; }
    public string Title { get; init; } = string.Empty;
    public string? Description { get; init; }
    public int AuthorId { get; init; }
    public int CurrentAssigneeId { get; init; }
    public int? NextAssigneeId { get; init; }
    public TaskStatus Status { get; init; }
    public DateTime? DueDate { get; init; }
    public DateTime? CompletedAt { get; init; }
    public DateTime CreatedAt { get; init; }
    public DateTime UpdatedAt { get; init; }
}

File: collab_todo/models/task.py
"""업무 도메인 모델."""

from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime
from typing import Optional, Literal


TaskStatus = Literal[
    "pending",
    "confirmed",
    "in_progress",
    "review",
    "completed",
    "on_hold",
    "cancelled",
]


@dataclass(frozen=True)
class Task:
    id: int
    project_id: int
    parent_task_id: Optional[int]
    title: str
    description: Optional[str]
    author_id: int
    current_assignee_id: int
    next_assignee_id: Optional[int]
    status: TaskStatus
    due_date: Optional[datetime]
    confirmed_at: Optional[datetime]
    promised_date: Optional[datetime]
    is_private: bool
    completed_at: Optional[datetime]
    created_at: datetime
    updated_at: datetime
